processing/src/main/java/io/druid/query/lookup/LookupExtractionFn.java

```diff
@@ -90,7 +90,8 @@ public boolean isInjective()
   @JsonProperty("optimize")
   public boolean isOptimize()
   {
-    return optimize;
+    // case retainMissingValue == true  we can not optimize need to push down the filter
+    return !isRetainMissingValue() && optimize;
```

Comment:
@b-slim I think this should be the in the optimize() call, it's the same kind of check as ```+      if (Strings.emptyToNull(lookupExtractionFn.getReplaceMissingValueWith()) == Strings.emptyToNull(value)) {
``` and I think that should go in the optimize() call itself